starport-core/src/main/scala/com/krux/starport/metric/MetricSettings.scala

```diff
@@ -3,38 +3,107 @@ package com.krux.starport.metric
+      case CloudWatchReporterSettings => CloudWatchReporterSettings(config).getReporter(registry)
+  }
 
-  def getReporter(registry: MetricRegistry): ScheduledReporter = GraphiteReporter.forRegistry(registry)
-    .prefixedWith(metricPrefix)
-    .convertDurationsTo(MetricSettings.DefaultDuration)
-    .build(getGraphite)
+  def getDefaultReporter(registry: MetricRegistry): ScheduledReporter = ConsoleReporter
+    .forRegistry(registry)
+    .convertDurationsTo(DefaultDuration)
+    .build()
 
 }
 
-object MetricSettings {
+final case object GraphiteReporterSettings extends MetricSettings {
 
-  final val DefaultDuration = TimeUnit.SECONDS
+  def apply(config: Config): MetricSettingsImpl = GraphiteReporterSettingsImpl(config)
+
+  private case class GraphiteReporterSettingsImpl(config: Config) extends MetricSettingsImpl {
+
+    val hosts = config.getStringList("hosts")
+
+    val port = config.getInt("port")
+
+    val metricPrefix = config.getString("prefix")
+
+    def getGraphite: Graphite =
+      new Graphite(hosts.get(Random.nextInt(hosts.size)), port)
+
+    def getReporter(registry: MetricRegistry): ScheduledReporter = GraphiteReporter
+      .forRegistry(registry)
+      .prefixedWith(metricPrefix)
+      .convertDurationsTo(DefaultDuration)
+      .build(getGraphite)
+
+  }
 
-  def getReporter(configOpt: Option[Config], registry: MetricRegistry): ScheduledReporter =
-    configOpt.map(new MetricSettings(_).getReporter(registry))
-      .getOrElse(
-        ConsoleReporter.forRegistry(registry).convertDurationsTo(DefaultDuration).build()
-      )
 }
+
+final case object CloudWatchReporterSettings extends MetricSettings {
+
+  def apply(config: Config): MetricSettingsImpl = CloudWatchReporterSettingsImpl(config)
+
+  private case class CloudWatchReporterSettingsImpl(config: Config) extends MetricSettingsImpl {
+
+    val awsRegion: Region = Region.of(config.getString("region"))
+    val awsEnvironment = config.getString("environment")
+
+    val awsCloudWatchAsync: CloudWatchAsyncClient = CloudWatchAsyncClient
+      .builder
+      .region(awsRegion)
+      .build
+
+    def getReporter(metricRegistry: MetricRegistry): CloudWatchReporter = CloudWatchReporter
+      .forRegistry(metricRegistry, awsCloudWatchAsync, classOf[Nothing].getName)
+      .convertRatesTo(TimeUnit.SECONDS).convertDurationsTo(TimeUnit.MILLISECONDS)
+      .filter(MetricFilter.ALL)
+      .withPercentiles(Percentile.P75, Percentile.P99)
+      .withOneMinuteMeanRate
+      .withFiveMinuteMeanRate
+      .withFifteenMinuteMeanRate
+      .withMeanRate
+      .withArithmeticMean
+      .withStdDev
+      .withStatisticSet
+      .withZeroValuesSubmission
+      .withReportRawCountValue
+      .withHighResolution
+      .withMeterUnitSentToCW(StandardUnit.BYTES)
+      .withJvmMetrics
+      .withGlobalDimensions(s"Region=${awsRegion}", s"Instance=${awsEnvironment}")
+      .withDryRun
```

Comment:
what does `withDryRun` do?

Reply:
Ah thanks, that's a good spot. It will just `log.DEBUG` the output rather than posting to CloudWatch. Testing only.

Reply:
Consider defaulting to `ConsoleReporter` (https://metrics.dropwizard.io/3.1.0/getting-started/)

Reply:
So I wrote it a little differently. The `MetricSettings` trait has a default method which will construct a `ConsoleReporter`. Then, in the code where we actually call an instance such as `StartScheduledPipelines`, we try to call the instance and if it's unsuccessful we call the default `ConsoleReporter`. 

Have a look at lines 267 - 270 in `StartScheduledPipelines` and 161 - 164 in `CleanupExistingPipelines`. I am open to moving this within the constructor if that's preferred or has benefits I may not have thought of.

Reply:
Seems like a reasonable fallback then, great.